### DISCUSSION TITLE
is there a after effect for x-for template once it rendered

### DISCUSSION THREAD

**Discussion by skandasoft** [alpinejs/alpine#2197] [1. Help] [answered]
is there a after effect for x-for template once it rendered. Let say if it the first item in the template rendered need to have the focus on it.

**Comment by SimoTod** [comment, marked as answer]
There's nothing specific for x-for but I think you can use x-effect + nextTick `<template x-for="item in items" x-effect="items && nextTick(() => ...)>`

**Reply by skandasoft** [reply]
tried it..didn't go well..the x-effect function was called even before the template is complete and result shown(i was using a alert)

**Reply by SimoTod** [reply]
```html 
<div x-data="{items:['one', 'two']}">
  <template x-for="item in items" x-effect="items && $nextTick(() => $root.querySelectorAll('input')[0].focus())">
    <input :value="item">
  </template>
</div>
```
Seems to work for me.

**Reply by skandasoft** [reply]
thanks..let me try..

**Comment by noam-elbaz** [comment]
yup.  i've been trying alot different hacks and attempts to catch the item when its being rendered and then giving it some transitions from tailwind. but little success.  if anybody has any good ideas let me know.
